ssl/tls13_enc.c

```diff
@@ -797,9 +798,27 @@ int tls13_change_cipher_state(SSL *s, int which)
     if (s->record_padding_cb != NULL)
         goto skip_ktls;
 
+#  ifdef __FreeBSD__
+    memset(&crypto_info, 0, sizeof(crypto_info));
+    switch (s->s3.tmp.new_cipher->algorithm_enc) {
+    case SSL_AES128GCM:
+    case SSL_AES256GCM:
+        crypto_info.cipher_algorithm = CRYPTO_AES_NIST_GCM_16;
+        crypto_info.iv_len = EVP_CIPHER_CTX_iv_length(ciph_ctx);
+        break;
+    default:
+        goto skip_ktls;
+    }
+    crypto_info.cipher_key = key;
+    crypto_info.cipher_key_len = EVP_CIPHER_key_length(cipher);
+    crypto_info.iv = iv;
+    crypto_info.tls_vmajor = (s->version >> 8) & 0x000000ff;
+    crypto_info.tls_vminor = (s->version & 0x000000ff);
```

Comment:
This block seems to do a similar thing to the `ktls_check_supported_cipher` and `ktls_configure_crypto` functions. It seems like it would be better to modify those functions to do this, rather than insert this block here.

Reply:
So I've actually posted a new PR #12111 which does do that, but with some caveats as FreeBSD and Linux need some slightly different arguments.  I also raised the question about a typedef in that review as well.  The new PR also includes #11679 since the diffs for both of these ended up much smaller after doing the refactoring so that I think they are ok to review together.  Apologies if that wasn't the best way to do that?